openam-oauth2/src/main/java/org/forgerock/oauth2/core/ResourceOwnerSessionValidator.java

```diff
@@ -311,25 +314,53 @@ private long getMaxAge(OAuth2Request request)
     private void setCurrentAcr(SSOToken token, OAuth2Request request, String acrValuesStr)
             throws NotFoundException, ServerException, SSOException, AccessDeniedException,
             UnsupportedEncodingException, URISyntaxException, ResourceOwnerAuthenticationRequired {
-        String serviceUsed = token.getProperty(ISAuthConstants.SERVICE);
+    	 Set<String> serviceUsedSet = parseServiceProperty(token.getProperty(ISAuthConstants.SERVICE));
         Set<String> acrValues = new HashSet<>(Arrays.asList(acrValuesStr.split("\\s+")));
         OAuth2ProviderSettings settings = providerSettingsFactory.get(request);
         Map<String, AuthenticationMethod> acrMap = settings.getAcrMapping();
         final Request req = request.getRequest();
         boolean matched = false;
-        for (String acr : acrValues) {
-            if (acrMap.containsKey(acr)) {
-                if (serviceUsed.equals(acrMap.get(acr).getName())) {
-                    req.getResourceRef().addQueryParameter(OAuth2Constants.JWTTokenParams.ACR, acr);
-                    matched = true;
-                    break;
-                }
-            }
+        boolean containsValidAcr = false;
+        String matchedAcr = "0";
```

Comment:
You should use `UNMATCHED_ACR_VALUE` here.

Reply:
I use UNMATCHED_ACR_VALUE insted of "0".